src/Shared/FileSystem/CachingFileSystemWrapper.cs

```diff
@@ -15,10 +15,17 @@ internal sealed class CachingFileSystemWrapper : IFileSystem
         private readonly IFileSystem _fileSystem;
         private readonly ConcurrentDictionary<string, bool> _existenceCache = new ConcurrentDictionary<string, bool>();
         private readonly ConcurrentDictionary<string, DateTime> _lastWriteTimeCache = new ConcurrentDictionary<string, DateTime>();
+        private readonly bool _skipExistenceCheck = false;
 
         public CachingFileSystemWrapper(IFileSystem fileSystem)
+            : this(fileSystem, false)
+        {
+        }
+
+        public CachingFileSystemWrapper(IFileSystem fileSystem, bool skipExistenceCheck = false)
         {
             _fileSystem = fileSystem;
+            _skipExistenceCheck = skipExistenceCheck;
```

Comment:
Would it be even better that if `skipExistenceCheck` was `true`, it never creates `_existenceCache` and the `Exists()` methods just return true?

Reply:
yes, that works, good call